infrastructure.md

```diff
@@ -8,4 +8,100 @@ The best description of the full infrastructure is currently given as an [Ansibl
 
 The plans for the final set-up are given in [this presentation Jonathan prepared for Harmony 2018](https://github.com/ModellingWebLab/WLDocs/blob/master/doc/WL2%20technical%20detail.pdf)
 
+## Getting started
+
+To get started practically, jump down to the deployment section **TODO: ADD LINK**
+
+## Overview
+
+This diagram describes the general purpose Web Lab infrastructure.
+
+![schematic overview](img/infra-1.png)
+
+## Django website
+
+The website is written in Python using [Django](https://docs.djangoproject.com/en/2.2/).
+The Django server serves up a HTML and javascript front-end, which can interact with the Django server backend via a REST API.
+**JONATHAN: DID WE ADD ANYTHING HERE, OR IS THIS ALL HANDLED BY DJANGO?**
+
+The current Django application is slightly cardiac-specific, and is hosted in the [WebLab](https://github.com/ModellingWebLab/WebLab) repository.
+
+The django back-end can access data from various sources (databases, local files, git repositories, a metadata triple store), and can communicate with the "task queue" to execute experiments via a REST API.
+**JONATHAN: IS THIS API DOCUMENTED SOMEWHERE?**
+
```

Comment:
What are `nginx` and `uwsgi` and where do they come in? Are they part of the django playbook?

Reply:
Yes. `nginx` is a web server (think equivalent of Apache but more lightweight). `uwsgi` is a WSGI server, which serves WSGI apps (Django in our case) - think faster CGI. Requests come in to `nginx` which responds directly if a static resource, or hands off to Django via `uwsgi` if not.

The back-end fc-runner is also served via `nginx`, but this time it's an old-fashioned CGI app wrapped via `fcgiwrap`, since `nginx` doesn't do old-fashioned CGI natively.